s3_storage_client.go

```diff
@@ -87,8 +87,11 @@ func (util *snowflakeS3Client) getFileHeader(meta *fileMetadata, filename string
 			}
 			meta.resStatus = errStatus
 			meta.lastError = err
-			return nil, fmt.Errorf("error while retrieving header")
+			return nil, fmt.Errorf("s3 api error while retrieving header")
 		}
+		meta.resStatus = errStatus
+		meta.lastError = err
+		return nil, fmt.Errorf("unexpected error while retrieving header")
```

Comment:
I have a strong belief, that go devs depend more on this error than on the `lastError` field. So IMHO it would be better to include `err` in return value. We can either return just `return nil, err` or add a message like `return nil, fmt.Errorf("unexpected error while retrieving header. %v", err)` (not sure about syntax, please check it :) )

Reply:
good point, changed